frontend/src/presentation/components/QuestionFilters.tsx

```diff
@@ -0,0 +1,181 @@
+    },
+    []
+  );
+
+  const handleCategoryTagChange = useCallback(
+    (category: string, checked: boolean) => {
+      const nextSelectedCategories = checked
+        ? [...selectedCategories, category]
+        : selectedCategories.filter((c) => c !== category);
+      setSelectedCategories(nextSelectedCategories);
+      triggerFiltersChange({
+        selectedDifficulty,
+        selectedCategories: nextSelectedCategories,
+        searchTerm,
+      });
+    },
+    [selectedDifficulty, selectedCategories, searchTerm, triggerFiltersChange]
+  );
+
+  const filteredCategories = useMemo(() => {
+    const lowerSearchTerm = categorySearchTerm.toLowerCase();
+    return allCategories.filter((category) =>
+      category.toLowerCase().includes(lowerSearchTerm)
+    );
+  }, [allCategories, categorySearchTerm]);
+
+  const categoryDropdownContent = (
+    <div
+      className={styles.categoryDropdownContent}
+      onClick={(e) => e.stopPropagation()}
+    >
+      <div className={styles.categoryDropdownContainer}>
+        <Input
+          placeholder="Search Categories"
+          value={categorySearchTerm}
+          onChange={handleCategorySearch}
+          className={styles.categorySearchBar}
+          allowClear
+        />
+        <div className={styles.categoriesGrid}>
+          {filteredCategories.map((category) => (
+            <CheckableTag
+              key={category}
+              checked={selectedCategories.includes(category)}
+              onChange={(checked) => handleCategoryTagChange(category, checked)}
+            >
+              {category}
+            </CheckableTag>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+
+  return (
+    <div className={styles.filtersContainer}>
+      <div className={styles.filtersRow}>
+        <div className={styles.filterItem}>
+          <Dropdown
+            overlay={categoryDropdownContent}
+            trigger={['click']}
+            overlayClassName={styles.categoryDropdownOverlay}
+          >
+            <Button className={styles.categoriesFilterButton}>
+              Categories <DownOutlined />
+            </Button>
+          </Dropdown>
+        </div>
+
+        <div className={styles.filterItem}>
+          <Select
+            placeholder="Select Difficulty"
+            value={selectedDifficulty}
+            onChange={handleDifficultyChange}
+            className={styles.difficultyFilter}
+            optionLabelProp="label"
+          >
+            <Option value="All" label="All">
+              All
+            </Option>
+            <Option
+              value="Easy"
+              label={<span style={{ color: 'green' }}>Easy</span>}
+            >
+              <span style={{ color: 'green' }}>Easy</span>
+            </Option>
+            <Option
+              value="Medium"
+              label={<span style={{ color: 'goldenrod' }}>Medium</span>}
+            >
+              <span style={{ color: 'goldenrod' }}>Medium</span>
```

Comment:
Lol I didnt know such a colour exists. Cool

Reply:
I just straight up didn't see that that is the deprecated version lol.